packages/gateway-client/src/worker/p2p-client/stream-factory.ts

```diff
@@ -139,4 +222,27 @@ export class StreamFactory {
     public async getStream(protocol = '/samizdapp-proxy') {
         return this.makeStream(protocol);
     }
+
+    public async getProxyStream() {
+        do {
+            const stream = this.availableStreams.pop();
+            if (stream?.isOpen) {
+                return stream;
+            }
+        } while (this.availableStreams.length > 0);
+
+        const wrapped = new WrappedStream(
+            await this.getStream('/samizdapp-proxy/2.0.0')
+        );
+        wrapped.open();
+        return wrapped;
+    }
+
+    public releaseProxyStream(stream: WrappedStream) {
+        this.availableStreams.push(stream);
+    }
+
+    public closeStream(stream: WrappedStream) {
```

Comment:
Why do we need this method instead of just calling `close()` on the stream directly?